infrastructure/src/main/java/org/wso2/testgrid/infrastructure/providers/aws/utils/AwsAMIMapper.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.testgrid.infrastructure.providers.aws.utils;
+
+import com.amazonaws.auth.PropertiesFileCredentialsProvider;
+import com.amazonaws.services.ec2.AmazonEC2;
+import com.amazonaws.services.ec2.AmazonEC2ClientBuilder;
+import com.amazonaws.services.ec2.model.DescribeImagesRequest;
+import com.amazonaws.services.ec2.model.DescribeImagesResult;
+import com.amazonaws.services.ec2.model.Image;
+import com.amazonaws.services.ec2.model.Tag;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.testgrid.common.exception.TestGridInfrastructureException;
+import org.wso2.testgrid.common.util.StringUtil;
+import org.wso2.testgrid.common.util.TestGridUtil;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * This class provides a mapper to find the matching AMI based on the infrastructure parameters of the test-plan.
+ */
+public class AwsAMIMapper {
+    private static final String AWS_REGION_NAME = "us-east-1";
+    private static final Logger logger = LoggerFactory.getLogger(AwsAMIMapper.class);
+
+    private final AmazonEC2 amazonEC2;
+
+    public AwsAMIMapper() throws IOException, TestGridInfrastructureException {
+        Path configFilePath = Paths.get(TestGridUtil.getTestGridHomePath(), "config.properties");
+        if (!configFilePath.toFile().exists()) {
```

Comment:
You could use "Files.exists(configFilePath)" directly. In the meantime, it would be better to check if it is a directory prior to proceeding.